api/payments.go

```diff
@@ -276,18 +284,17 @@ func (a *API) PaymentRefund(ctx context.Context, w http.ResponseWriter, r *http.
 		return
 	}
 
-	provider := gcontext.GetPaymentProvider(ctx)
+	provider := gcontext.GetPaymentProviders(ctx)[order.PaymentProcessor]
```

Comment:
I think that there are a few instances when we'd want to empty/nil check the value in this PR. 

By normalizing, if we are taking the name from the config it might be `Stripe` and we should make it work on `stripe`